kubernetes/cache/istio.go

```diff
@@ -11,807 +10,536 @@ import (
-}
-
-func (c *kialiCacheImpl) isIstioSynced(namespace string) bool {
-	var isSynced bool
-	if nsCache, exist := c.nsCache[namespace]; exist {
-		isSynced = true
-		if c.CheckIstioResource(kubernetes.DestinationRules) {
-			isSynced = isSynced && nsCache[kubernetes.DestinationRuleType].HasSynced()
-		}
-		if c.CheckIstioResource(kubernetes.EnvoyFilters) {
-			isSynced = isSynced && nsCache[kubernetes.EnvoyFilterType].HasSynced()
-		}
-		if c.CheckIstioResource(kubernetes.Gateways) {
-			isSynced = isSynced && nsCache[kubernetes.GatewayType].HasSynced()
-		}
-		if c.CheckIstioResource(kubernetes.ServiceEntries) {
-			isSynced = isSynced && nsCache[kubernetes.ServiceEntryType].HasSynced()
-		}
-		if c.CheckIstioResource(kubernetes.Sidecars) {
-			isSynced = isSynced && nsCache[kubernetes.SidecarType].HasSynced()
-		}
-		if c.CheckIstioResource(kubernetes.VirtualServices) {
-			isSynced = isSynced && nsCache[kubernetes.VirtualServiceType].HasSynced()
-		}
-		if c.CheckIstioResource(kubernetes.WorkloadGroups) {
-			isSynced = isSynced && nsCache[kubernetes.WorkloadGroupType].HasSynced()
-		}
-		if c.CheckIstioResource(kubernetes.WorkloadEntries) {
-			isSynced = isSynced && nsCache[kubernetes.WorkloadEntryType].HasSynced()
-		}
-
-		if c.CheckIstioResource(kubernetes.AuthorizationPolicies) {
-			isSynced = isSynced && nsCache[kubernetes.AuthorizationPoliciesType].HasSynced()
-		}
-		if c.CheckIstioResource(kubernetes.PeerAuthentications) {
-			isSynced = isSynced && nsCache[kubernetes.PeerAuthenticationsType].HasSynced()
-		}
-		if c.CheckIstioResource(kubernetes.RequestAuthentications) {
-			isSynced = isSynced && nsCache[kubernetes.RequestAuthenticationsType].HasSynced()
-		}
-	} else {
-		isSynced = false
-	}
-	return isSynced
+	return sharedInformers
 }
 
 func (c *kialiCacheImpl) GetDestinationRule(namespace, name string) (*networking_v1beta1.DestinationRule, error) {
 	if !c.CheckIstioResource(kubernetes.DestinationRules) {
 		return nil, fmt.Errorf("Kiali cache doesn't support [resourceType: %s]", kubernetes.DestinationRuleType)
 	}
-	if nsCache, ok := c.nsCache[namespace]; ok {
-		// Cache stores natively items with namespace/name pattern, we can skip the Indexer by name and make a direct call
-		key := namespace + "/" + name
-		obj, exist, err := nsCache[kubernetes.DestinationRuleType].GetStore().GetByKey(key)
-		if err != nil {
-			return nil, err
-		}
-		if exist {
-			l, ok := obj.(*networking_v1beta1.DestinationRule)
-			if !ok {
-				return nil, errors.New("bad DestinationRule type found in cache")
-			}
-			// Informers don't always populate Kind field
-			l.Kind = kubernetes.DestinationRuleType
-			log.Tracef("[Kiali Cache] Get [resource: DestinationRule] for [namespace: %s] [name: %s]", namespace, name)
-			return l, nil
-		}
-	}
-	return nil, nil
+
+	dr, err := c.getCacheLister(namespace).destinationRuleLister.DestinationRules(namespace).Get(name)
+	if err != nil {
+		return nil, err
+	}
+
+	dr.Kind = kubernetes.DestinationRuleType
```

Comment:
Ah, nice.
Yes, k8s apimachinery returns "empty" and "Kind" and those are used when parsing objects into YAML into the frontend.